core/services/remote_access_services.py

```diff
@@ -121,9 +121,11 @@ def store_trained_classifier_model(job_result_dict):
     if 'classifier_data' not in job_result_dict:
         raise Exception('job_result_dict must contain \'classifier_data\'.')
 
-    payload = job_result_dict
-    payload['vm_id'] = _get_vm_id()
-    signature = generate_signature(payload)
+    payload = {
+        'message': job_result_dict,
+        'vm_id': _get_vm_id()
+    }
+    signature = generate_signature(payload['message'])
     payload['signature'] = signature
```

Comment:
@anmolshkl We could do that. But then, signature is meant to be an encoding of the message we pass to the controller and the message is a key in the payload dict. It doesn't really seem to point to anything other than the message dict.